### PR TITLE
Migrate some container copy tests to integration

### PR DESCRIPTION
**Pull request by dnephin**
Fixes https://jenkins.dockerproject.org/job/docker-ce/257/execution/node/465/log/?consoleFull

cc @andrewhsu

### CODE REVIEW

File: integration-cli/docker_cli_cp_from_container_test.go [hunk 1]
```diff
@@ -20,67 +18,9 @@ import (
 // 3. DST parent directory must exist.
 // 4. If DST exists as a file, it must not end with a trailing separator.
 
-// First get these easy error cases out of the way.
-
-// Test for error when SRC does not exist.
-func (s *DockerSuite) TestCpFromErrSrcNotExists(c *check.C) {
```
Comment (dnephin) [inline integration-cli/docker_cli_cp_from_container_test.go]: Moved to `TestCopyFromContainerPathDoesNotExist`

File: integration-cli/docker_cli_cp_from_container_test.go [hunk 2]
```diff
@@ -20,67 +18,9 @@ import (
...
-
-	err := runDockerCp(c, containerCpPath(containerID, "file1"), tmpDir, nil)
-	c.Assert(err, checker.NotNil)
-
-	c.Assert(isCpNotExist(err), checker.True, check.Commentf("expected IsNotExist error, but got %T: %s", err, err))
-}
-
-// Test for error when SRC ends in a trailing
-// path separator but it exists as a file.
-func (s *DockerSuite) TestCpFromErrSrcNotDir(c *check.C) {
```
Comment (dnephin) [inline integration-cli/docker_cli_cp_from_container_test.go]: Moved to `TestCopyFromContainerPathIsNotDir`

File: integration-cli/docker_cli_cp_from_container_test.go [hunk 3]
```diff
@@ -20,67 +18,9 @@ import (
...
-
-	err := runDockerCp(c, containerCpPathTrailingSep(containerID, "file1"), tmpDir, nil)
-	c.Assert(err, checker.NotNil)
-
-	c.Assert(isCpNotDir(err), checker.True, check.Commentf("expected IsNotDir error, but got %T: %s", err, err))
-}
-
-// Test for error when DST ends in a trailing
-// path separator but exists as a file.
-func (s *DockerSuite) TestCpFromErrDstNotDir(c *check.C) {
```
Comment (dnephin) [inline integration-cli/docker_cli_cp_from_container_test.go]: Exists in `docker/cli` as [TestRunCopyFromContainerToFilesystemMissingDestinationDirectory](https://github.com/docker/cli/blob/master/cli/command/container/cp_test.go#L92)

File: integration-cli/docker_cli_cp_to_container_test.go [hunk 4]
```diff
@@ -21,124 +17,6 @@ import (
 // 3. DST parent directory must exist.
 // 4. If DST exists as a file, it must not end with a trailing separator.
 
-// First get these easy error cases out of the way.
-
-// Test for error when SRC does not exist.
-func (s *DockerSuite) TestCpToErrSrcNotExists(c *check.C) {
```
Comment (dnephin) [inline integration-cli/docker_cli_cp_to_container_test.go]: Added in https://github.com/docker/cli/pull/907 as `TestRunCopyToContainerSourceDoesNotExist`

File: integration-cli/docker_cli_cp_to_container_test.go [hunk 5]
```diff
@@ -21,124 +17,6 @@ import (
...
-		c.Assert(strings.ToLower(err.Error()), checker.Contains, "cannot find the file specified")
-		c.Assert(strings.ToLower(err.Error()), checker.Contains, strings.ToLower(tmpDir))
-	} else {
-		c.Assert(strings.ToLower(err.Error()), checker.Contains, strings.ToLower(srcStatErr.Error()))
-	}
-}
-
-// Test for error when SRC ends in a trailing
-// path separator but it exists as a file.
-func (s *DockerSuite) TestCpToErrSrcNotDir(c *check.C) {
```
Comment (dnephin) [inline integration-cli/docker_cli_cp_to_container_test.go]: Added in https://github.com/docker/cli/pull/907 as `TestRunCopyToContainerFromFileWithTrailingSlash`

File: integration-cli/docker_cli_cp_to_container_test.go [hunk 6]
```diff
@@ -21,124 +17,6 @@ import (
...
-
-	err := runDockerCp(c, srcPath, dstPath, nil)
-	c.Assert(err, checker.NotNil)
-
-	c.Assert(isCpNotDir(err), checker.True, check.Commentf("expected IsNotDir error, but got %T: %s", err, err))
-}
-
-// Test for error when SRC is a valid file or directory,
-// but the DST parent directory does not exist.
-func (s *DockerSuite) TestCpToErrDstParentNotExists(c *check.C) {
```
Comment (dnephin) [inline integration-cli/docker_cli_cp_to_container_test.go]: Moved to `TestCopyToContainerPathDoesNotExist`

File: integration-cli/docker_cli_cp_to_container_test.go [hunk 7]
```diff
@@ -21,124 +17,6 @@ import (
...
-	err = runDockerCp(c, srcPath, dstPath, nil)
-	c.Assert(err, checker.NotNil)
-
-	c.Assert(isCpNotExist(err), checker.True, check.Commentf("expected IsNotExist error, but got %T: %s", err, err))
-}
-
-// Test for error when DST ends in a trailing path separator but exists as a
-// file. Also test that we cannot overwrite an existing directory with a
-// non-directory and cannot overwrite an existing
-func (s *DockerSuite) TestCpToErrDstNotDir(c *check.C) {
```
Comment (dnephin) [inline integration-cli/docker_cli_cp_to_container_test.go]: Moved to `TestCopyToContainerPathIsNotDir`

**Comment by arm64b** [comment]
`powerpc` complains something like `no space left` or `build timeout`, so I think we need to take action to make sure this node is reliable to judge the PR, especially for those ones which owner is not maintainer (since they have no right to `rebuild` it), not fair enough 🉑

**Review by vdemeester — APPROVED** [review]
LGTM :tiger:

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**